src/router/graphql.ts

```diff
@@ -36,14 +37,21 @@ if (config.graphql) {
             externalUrlBase: `/v1/graphql/${db}`,
             graphiql: withGraphiql,
             watchPg: true,
+            retryOnInitFail: (_error, _attempts) => {
+              postgraphiles[db].graphile.shutdown();
+              return false;
+            },
```

Comment:
This part is the actual fix. If you don't define `retryOnInitFailure` it simply shuts down the server, which is insane!

Reply:
what is doing when reaching the problem? just killed the worker? Does it get in a weird state? is it returning some kind of error?

Reply:
The init failure is about the schema generation failing. This function returns `false` meaning "don't continue retrying the initialization process and it calls `shutdown` so the graphile process just always returns query failures. Apparently it returns a `503`. Hopefully that doesn't raise the ire of Fargate.

![Screenshot from 2022-08-03 11-06-27](https://user-images.githubusercontent.com/765551/182656502-3fb4c0ce-12f1-4c3f-950d-364a055f1154.png)

Reply:
should we test it on staging?